controllers/auth.js

```diff
@@ -0,0 +1,51 @@
+const User = require('../models/User')
+const { StatusCodes } = require('http-status-codes')
+const { BadRequestError, UnauthenticatedError } = require('../errors/index')
+
+const register = async (req, res) => {
+    try {
+        const user = await User.create({...req.body}) // hashing pswd using mongoose pre method
+        // TODO: is it necessary to generate token during the registration
```

Comment:
It's not strictly necessary but it would be weird for a user to register and then need to separately go to the login page to get a token. 

Some systems do this, though. In particular, some systems will refuse to authenticate the user until they've validated their email, so after they register, they'll be redirected to a landing page asking them to validate their email instead of issuing a token.